dvc/tree/gdrive.py

```diff
@@ -510,19 +509,29 @@ def _path_to_item_ids(self, path, create, use_cache):
         )
 
     def _get_item_id(self, path_info, create=False, use_cache=True, hint=None):
-        assert path_info.bucket == self._bucket
-
-        item_ids = self._path_to_item_ids(path_info.path, create, use_cache)
-        if item_ids:
-            return min(item_ids)
+        if isinstance(path_info, GDriveURLInfo):
+            return path_info.host
+        else:
+            assert path_info.bucket == self._bucket
+            item_ids = self._path_to_item_ids(
+                path_info.path, create, use_cache
+            )
+            if item_ids:
+                return min(item_ids)
 
         assert not create
         raise FileMissingError(path_info, hint)
 
     def exists(self, path_info, use_dvcignore=True):
+        from pydrive2.files import ApiRequestError
+
         try:
-            self._get_item_id(path_info)
-        except FileMissingError:
+            item_id = self._get_item_id(path_info)
+            if isinstance(path_info, GDriveURLInfo):
+                file_repr = self._drive.CreateFile({"id": item_id})
+                file_repr.FetchMetadata()
+
+        except (FileMissingError, ApiRequestError):
```

Comment:
ApiRequestError -> `false` transition doesn't look right to me, ApiRequestError might mean pretty much everything, probably we even have some retry logic that will be broken by this - need to double check this

Reply:
Good point! I'll check a more precise form to check the file doesn't exist.